packages/nextjs/services/database/config/postgresClient.ts

```diff
@@ -4,26 +4,66 @@ import { drizzle as drizzleNeon } from "drizzle-orm/neon-serverless";
 import { drizzle } from "drizzle-orm/node-postgres";
 import { Pool } from "pg";
 
-let db: ReturnType<typeof drizzle<typeof schema>> | ReturnType<typeof drizzleNeon<typeof schema>>;
-
-const NEON_DB_STRING = "neondb";
-if (process.env.POSTGRES_URL?.includes(NEON_DB_STRING)) {
-  const neonPool = new NeonPool({ connectionString: process.env.POSTGRES_URL as string });
-  db = drizzleNeon(neonPool, { schema, casing: "snake_case" });
-} else {
-  const pool = new Pool({
-    connectionString: process.env.POSTGRES_URL,
-  });
-
-  db = drizzle(pool, {
-    schema,
-    casing: "snake_case",
-  });
-
-  pool.on("error", err => {
-    console.error("Unexpected error on idle client", err);
-    process.exit(-1);
-  });
+type DbInstance = ReturnType<typeof drizzle<typeof schema>> | ReturnType<typeof drizzleNeon<typeof schema>>;
+
+let dbInstance: DbInstance | null = null;
+let poolInstance: Pool | null = null;
+
+function getDb(): DbInstance {
+  if (dbInstance) {
+    return dbInstance;
+  }
+
+  const NEON_DB_STRING = "neondb";
+  if (process.env.POSTGRES_URL?.includes(NEON_DB_STRING)) {
+    const neonPool = new NeonPool({ connectionString: process.env.POSTGRES_URL as string });
+    dbInstance = drizzleNeon(neonPool, { schema, casing: "snake_case" });
+  } else {
+    const pool = new Pool({
+      connectionString: process.env.POSTGRES_URL,
+    });
+
+    poolInstance = pool;
+
+    dbInstance = drizzle(pool, {
+      schema,
+      casing: "snake_case",
+    });
+
+    pool.on("error", err => {
+      console.error("Unexpected error on idle client", err);
+      process.exit(-1);
+    });
+  }
+
+  return dbInstance;
+}
+
+export async function closeDb(): Promise<void> {
+  if (poolInstance) {
+    await poolInstance.end();
+    poolInstance = null;
+    dbInstance = null;
+  }
 }
 
-export { db };
+// Create a proxy to intercept all property accesses and method calls
+const dbProxy = new Proxy(
+  {},
+  {
+    get: (target, prop) => {
```

Comment:
nit: we can add type safety here: `prop: keyof DbProxy` and remove `any` at line 60

Reply:
Nice, thanks! e25467bb222f